google/cloud/forseti/services/scanner/dao.py

```diff
@@ -110,6 +113,13 @@ def create(self, violations, inventory_index_id):
             """
             with self.violationmaker() as session:
                 for violation in violations:
+
+                    violation_hash = _create_violation_hash(
+                        violation.get('full_name'),
```

Comment:
You should set the default value if the data is not in the dictionary to a string for the string types and an empty dict for violation_data.

Reply:
Done!